src/eventHandlers/WorkerInitHandler.ts

```diff
@@ -30,9 +31,19 @@ export class WorkerInitHandler extends EventHandler<'workerInitRequest', 'worker
         });
 
         logColdStartWarning(channel);
-        const functionAppDirectory = msg.functionAppDirectory;
-        if (functionAppDirectory) {
-            await channel.updatePackageJson(functionAppDirectory);
+
+        if (msg.hostVersion) {
+            channel.hostVersion = msg.hostVersion;
+        }
+
+        if (msg.functionAppDirectory) {
+            await startApp(msg.functionAppDirectory, channel);
+        } else {
+            channel.log({
+                message: 'functionAppDirectory was undefined in workerInitRequest message. Skipping startup logic.',
+                level: LogLevel.Warning,

```

Comment:
As mentioned, this needs to be `LogLevel.Information` or `LogLevel.Debug` if you're going to include a message about this. Also, please make the message more user-friendly with less internal variable names. Aka:

> Skipping app start logic because no function app directory was provided during worker init

Also if you're going to do this, you might as well do it in functionEnvironmentReload as well to be consistent.